src/Umbraco.Core/Telemetry/SiteIdentifierService.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Core.Configuration;
+using Umbraco.Cms.Core.Configuration.Models;
+
+namespace Umbraco.Cms.Core.Telemetry
+{
+    /// <inheritdoc />
+    internal class SiteIdentifierService : ISiteIdentifierService
+    {
+        private readonly IOptionsMonitor<GlobalSettings> _globalSettings;
```

Comment:
we should have the property as `GlobalSettings` instead and then use the OnChange method on the IOptionsMonitor to update it.  Otherwise, we will touch all config sources on each request